### PR TITLE
Add option to output constexpr generated constant primitive values

### PR DESCRIPTION
**Pull request by AlaskanEmily**
Let me know if there's anything I should change, or any test cases which would be good to have which I might have missed.

### CODE REVIEW

File: src/bindgen/ir/constant.rs [hunk 1]
```diff
@@ -491,12 +491,22 @@ impl Constant {
...
+            false
+        };
+
+        if (config.constant.allow_static_const || allow_constexpr) && config.language == Language::Cxx {
+            if allow_constexpr {
+                out.write("constexpr ")
             } else {
-                out.write("const ");
+               out.write(if in_body { "inline " } else { "static " });
+                if let Type::ConstPtr(..) = self.ty {
```
Comment (emilio) [inline src/bindgen/ir/constant.rs]: Indentation here looks off.

File: tests/rust/constant_constexpr.toml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+[const]
+allow_constexpr = true
+allow_static_const = true
+
```
Comment (emilio) [inline tests/rust/constant_constexpr.toml]: nit: extra new-line?

File: tests/rust/constant_constexpr.rs [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+pub const CONSTANT_FLOAT32: f32 = 312.292;
+pub const DELIMITER: char = ':';
+pub const LEFTCURLY: char = '{';
+#[repr(C)]
+struct Foo {
+    x: i32,
+}
+
+pub const SomeFoo: Foo = Foo{ x: 99, };
+
```
Comment (emilio) [inline tests/rust/constant_constexpr.rs]: nit: Extra newline?

File: src/bindgen/ir/constant.rs (hunk 1 above)
Comment (AlaskanEmily) [reply src/bindgen/ir/constant.rs]: Fixed

File: tests/rust/constant_constexpr.toml (hunk 2 above)
Comment (AlaskanEmily) [reply tests/rust/constant_constexpr.toml]: Fixed

File: tests/rust/constant_constexpr.rs (hunk 3 above)
Comment (AlaskanEmily) [reply tests/rust/constant_constexpr.rs]: Fixed

**Comment by AlaskanEmily** [comment]
I've added a section to the docs.md as well. I'm not really sure how in-depth I should make it, so I kept it simple.

**Comment by emilio** [comment]
The build was red because of rustfmt, so I fixed that and pushed https://github.com/eqrion/cbindgen/commit/462871695d621a055d32644f9bbdb4f9f9cdb38e. Thank you!

**Pull request closed without merging**